test/gtest/common/mem_buffer.cc

```diff
@@ -276,7 +276,8 @@ void mem_buffer::copy_from(void *dst, const void *src, size_t length,
 bool mem_buffer::compare(const void *expected, const void *buffer,
                          size_t length, ucs_memory_type_t mem_type)
 {
-    if (UCP_MEM_IS_ACCESSIBLE_FROM_CPU(mem_type)) {
+    if ((mem_type == UCS_MEMORY_TYPE_HOST) ||
```

Comment:
does it mean that CUDA-managed memory shouldn't be dereferenced at all?
if no, then I guess XPMEM/TCP/MM (i.e. not RDMA-like transports) could use CUDA/RoCM-managed memory directly

Reply:
@dmitrygx it can be dereferenced.  in the tests, if I use the memcpy for pattern fill/check for cuda-amanaged, then the pages will be always in the CPU land, RDMA never resulting in data validation failure.  used cuda_copy to just move away the pages to GPU land so that RDMA can result in data validation.

Reply:
@bureddy I see, thank you!

Reply:
maybe add comment - don't access managed memory from CPU to avoid moving the pages from GPU to CPU during the test